code/modules/mob/mob_helpers.dm

```diff
@@ -462,6 +462,8 @@
 			else
 				user.visible_message("[user] has fixed some of the [dam ? "dents on" : "burnt wires in"] [H]'s [parse_zone(affecting.body_zone)].", \
 					"<span class='notice'>You fix some of the [dam ? "dents on" : "burnt wires in"] [H == user ? "your" : "[H]'s"] [parse_zone(affecting.body_zone)].</span>")
+			if(((affecting.brute_dam <= 0 && brute_heal) && ((!H.is_bleeding()) && H.has_mechanical_bleeding())) || (affecting.burn_dam <= 0 && burn_heal)) //might be bad to have this many expressions in a single if....
+				return FALSE //successful heal, but the target is at full health. Returns false to signal you can stop healing now
```

Comment:
Actually didn't catch this earlier somehow, but this feels like its overcomplicating this section of code for code elsewhere. Why not just check to see if they're healed before looping in the welding healing loop?

Reply:
it requires both a check outside the loop and inside the loop. I decided changing the helper proc because it is used in one other place, and the return signifies if they still have damages left.